Add note about why we require all ports in the curated registry be simultaneously installable. (#332)
microsoft/vcpkg-docs: vcpkg/contributing/maintainer-guide.md

[Homebrew's Maintainer Guidelines](https://docs.brew.sh/Maintainer-Guidelines), and
[Homebrew's Formula Cookbook](https://docs.brew.sh/Formula-Cookbook).

> ## Overall Registry Design Goals

Reviewer: ```suggestion
## Overall registry design goals
```
We should use sentence case for all headings.

Author: Also fixed a few other examples that were wron

> ### Ports in the current baseline must be simultaneiously installable

Reviewer: ```suggestion
### Ports in the current baseline must be installable simultaneously
```

> We wish to be able to show downstream customers of libraries in the default registry that the

Reviewer: ```suggestion
We wish to be able to show downstream users of libraries in the curated registry that the
```

Author: Billy: (Or at least, that is the intent and uses of 'default' to mean github.com/microsoft/vcpkg are bugs)

Also Billy: (this PR)

:sigh:

combination of libraries in any given baseline we publish have been tested to work together in at
least some configurations. Allowing ports to exclude each other breaks the ability to test such
configurations, as the number of builds necessary for such tests would grow as

> `2^number_of_such_cases`. Moreover, installing additional dependencies is always considered "safe":

Reviewer: Not sure what the term "2^number_of_such_cases" means here?

Author: If we have 1 such alternative, we need to build the universe twice. If there are 2, we need to build the universe 4 times. If there are 3, we need to build the universe 8 times. And so on.

Author: (2 raised to the power of the number of such alternatives)

there is no way for a port or end user to assert that a dependency is *not* installed in their
requirements.

> If you wish to represent such an alternative situation for customers, consider describing how

Reviewer: Suggestion: Use "users" rather than "customers" throughout this article since "customers" often implies "paying users".

someone can create an [overlay port](../concepts/overlay-ports.md) implementing the alternative
form with a comment in `portfile.cmake` rather than trying to add additional ports never built in
the curated registry's continuous integration. For example, see
[glad@0.1.36](https://github.com/microsoft/vcpkg/blob/67cc1677c3bf5c23ea14b9d2416c7422fdeac492/ports/glad/portfile.cmake#L17).

Before the introduction of [registries](../maintainers/registries.md), we accepted several
not tested ports-as-alternatives, such as `boringssl`, that could make authoring overlay ports
easier. This is no longer accepted because registries allow publishing of these untested ports
without modifying the curated registry.

## PR structure

> ### Make separate Pull Requests per port

Reviewer: ```suggestion
### Make separate pull requests per port
```
This wasn't changed by your PR, but looking at this, I don't think we need to capitalize "Pull Requests" when we spell out both words.
